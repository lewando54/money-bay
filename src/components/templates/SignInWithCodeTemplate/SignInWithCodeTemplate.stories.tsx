/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'
import Images from '@assets/images'
import SignInWithCodeTemplate from './SignInWithCodeTemplate'

const meta = {
    component: SignInWithCodeTemplate,
    argTypes: {
        name: {control: 'text'},
        subname: {control: 'text'},
        profilePicture: {control: 'file'},
        pin: {control: 'text'},
        onSubmit: {action: 'Submitted'},
        onChange: {action: 'PinButton Pressed'},
        onFaceidPress: {action: 'FaceidPress Pressed'},
        onClosePress: {action: 'ClosePress Pressed'},
        onLostPasswordPress: {action: 'LostPasswordPress Pressed'}
    }
} satisfies Meta<typeof SignInWithCodeTemplate>

export default meta
type Story = StoryObj<typeof SignInWithCodeTemplate>

export const Default: Story = {
    args: {
        name: 'Cristina Wolf',
        subname: 'Labore sunt',
        profilePicture: Images.profilePicture,
        pin: '',
    }
}
