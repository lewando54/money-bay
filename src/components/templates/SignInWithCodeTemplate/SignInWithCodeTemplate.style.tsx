import { Dimensions, StyleSheet } from 'react-native'
import globalThemeStyle, { BODY_TEXT_COLOR, MAIN_DARK } from '../../../styling/GlobalTheme.style'

const { width } = Dimensions.get('screen')

export default StyleSheet.create({
    background: {
        width: width,
        resizeMode: 'contain',
        maxHeight: width - 27,
        position: 'absolute'
    },
    mainContainer: {
        flex: 1,
        gap: 30,
        alignItems: 'center',
        justifyContent: 'center'
    },
    profilePicture: {
        borderRadius: 100,
        width: width * 0.1866,
        height: width * 0.1866
    },
    button: {
        width: 80,
        height: 70,
        alignItems: 'center',
        justifyContent: 'center'
    },
    buttonText: {
        ...globalThemeStyle.text_Regular,
        fontSize: 24,
        color: MAIN_DARK
    },
    buttonContainer: {
        width: width * 0.6933,
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'center',
        alignItems: 'center',
        gap: 10
    },
    usernameContainer: {
        alignItems: 'center',
        marginBottom: -10
    },
    usernameText: {
        ...globalThemeStyle.text_Medium,
        fontSize: 20,
        color: MAIN_DARK
    },
    usernameSubText: {
        ...globalThemeStyle.text_Medium,
        fontSize: 16,
        color: BODY_TEXT_COLOR
    }
})
