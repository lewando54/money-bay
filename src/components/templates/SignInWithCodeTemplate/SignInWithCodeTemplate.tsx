import React from 'react'
import signInWithCodeTemplateStyle from './SignInWithCodeTemplate.style'
import { View, Image, Text, ImageSourcePropType } from 'react-native'

import Close from './assets/ep_close.svg'
import FaceID from './assets/faceid-line.svg'
import Link from '../../atoms/Link/Link'
import Images from 'assets/images'
import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'
import PinDots from '../../molecules/PinDots/PinDots'
import { Easing, useSharedValue, withRepeat, withSequence, withTiming } from 'react-native-reanimated'

interface ISignInWithCodeTemplateProps {
    name?: string
    subname?: string
    profilePicture?: ImageSourcePropType
    pin?: string
    onSubmit?: (pin: string) => boolean
    onChange?: (newValue: string) => void
    onFaceidPress?: () => void
    onClosePress?: () => void
    onLostPasswordPress?: () => void
    testID?: string
}

export default function SignInWithCodeTemplate({
    name='Cristina Wolf',
    subname='Labore sunt',
    profilePicture=Images.profile,
    pin='',
    onSubmit,
    onChange,
    onFaceidPress,
    onClosePress,
    onLostPasswordPress,
    testID
}: ISignInWithCodeTemplateProps){
    const keypadArray = [
        1, 2, 3, 4, 5, 6, 7, 8, 9, 'faceid', 0, 'close'
    ]

    const x = useSharedValue(0)

    const OFFSET = 10
    const TIME = 100

    const onChangeHandle = (newValue: number) => {
        if(onChange){
            onChange(pin + newValue.toString())
        }
        if(pin.length == 3){
            setTimeout(() => {
                let isCorrect = false
                if(onSubmit){
                    isCorrect = onSubmit(pin + newValue.toString())
                }
                if(isCorrect){
                    console.log('correct')
                }
                else{
                    x.value = withSequence(
                        withTiming(-OFFSET, { duration: TIME / 2, easing: Easing.exp }),
                        withRepeat(withTiming(OFFSET, { duration: TIME, easing: Easing.exp }), 2, true),
                        withTiming(0, { duration: TIME / 2, easing: Easing.exp })
                    )
                    console.log('incorrect')
                }
                onChange('')
            }, 500)
        }
    }

    const Keypad = keypadArray.map((el, index) => {
        if(typeof el == 'number'){
            return <UniversalContainer 
                key={index}
                style={signInWithCodeTemplateStyle.button}
                castShadow={true}
                onPress={() => onChangeHandle(el)}>
                <Text style={signInWithCodeTemplateStyle.buttonText}>{el}</Text>
            </UniversalContainer>
        }
        else if(el === 'faceid'){
            return <UniversalContainer
                key={index}
                style={signInWithCodeTemplateStyle.button}
                castShadow={true}
                onPress={onFaceidPress}>
                <FaceID />
            </UniversalContainer>
        }
        else if(el === 'close'){
            return <UniversalContainer 
                key={index}
                style={signInWithCodeTemplateStyle.button}
                castShadow={true}
                onPress={onClosePress}>
                <Close />
            </UniversalContainer>
        }
    })

    return (
        <View testID={testID} style={{flex: 1}}>
            <Image source={Images.bgSignIn} style={signInWithCodeTemplateStyle.background} />
            <View style={signInWithCodeTemplateStyle.mainContainer}>
                <Image source={profilePicture} style={signInWithCodeTemplateStyle.profilePicture}/>
                <View style={signInWithCodeTemplateStyle.usernameContainer}>
                    <Text style={signInWithCodeTemplateStyle.usernameText}>{name}</Text>
                    <Text style={signInWithCodeTemplateStyle.usernameSubText}>{subname}</Text>
                </View>
                <PinDots currLength={pin.length} x={x}/>
                <View style={signInWithCodeTemplateStyle.buttonContainer}>
                    {Keypad}
                </View>
                <Link onPress={onLostPasswordPress}>Lost your password?</Link>
                <Link>Switch user</Link>
            </View>
        </View>
    )
}
