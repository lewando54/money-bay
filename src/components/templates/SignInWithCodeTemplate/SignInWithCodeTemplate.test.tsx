import React, { useState } from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

import SignInWithCodeTemplate from './SignInWithCodeTemplate'

interface IMockupProps {
    onSubmit?: (pin: string) => boolean
}

function Stateful({onSubmit}: IMockupProps) {
    const [pin, setPin] = useState('')
    const onChangeHandle = (newValue: string) => {
        setPin(() => newValue)
    }
    return (
        <SignInWithCodeTemplate onChange={onChangeHandle} onSubmit={onSubmit} pin={pin}/>
    )
}

describe('SignInWithCodeTemplate', () => {
    jest.useFakeTimers()
    it('should render correctly', () => {
        const tree = renderer.create(<SignInWithCodeTemplate />)
        expect(tree).toMatchSnapshot()
    })

    it('should enter the pin correctly', () => {
        const onChange = jest.fn()
        const { getByText } = render(<SignInWithCodeTemplate onChange={onChange} />)
        fireEvent.press(getByText('1'))
        expect(onChange).toBeCalledWith('1')
    })

    it('should call onSubmit when pin has been entered', async () => {
        const onSubmit = jest.fn()
        const { getByText } = render(<Stateful onSubmit={onSubmit}/>)
        fireEvent.press(getByText('1'))
        fireEvent.press(getByText('2'))
        fireEvent.press(getByText('3'))
        fireEvent.press(getByText('4'))
        setTimeout(() => expect(onSubmit).toBeCalledWith('1234'), 1000)
    })

    it('should not call onSubmit when wrong pin has been entered', async () => {
        const onSubmit = jest.fn()
        const { getByText } = render(<Stateful onSubmit={onSubmit}/>)
        fireEvent.press(getByText('1'))
        fireEvent.press(getByText('2'))
        fireEvent.press(getByText('3'))
        fireEvent.press(getByText('3'))
        setTimeout(() => expect(onSubmit).toBeCalledTimes(0), 1000)
    })
})
