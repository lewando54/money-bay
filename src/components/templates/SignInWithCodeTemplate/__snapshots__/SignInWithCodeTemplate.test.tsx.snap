// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInWithCodeTemplate should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "maxHeight": 723,
        "position": "absolute",
        "resizeMode": "contain",
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "gap": 30,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "borderRadius": 100,
          "height": 139.95,
          "width": 139.95,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": -10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishMedium",
              "fontSize": 20,
            },
          ]
        }
      >
        Cristina Wolf
      </Text>
      <Text
        style={
          [
            {
              "color": "#1B1D4D",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            {
              "color": "#4C4C60",
              "fontFamily": "MulishMedium",
              "fontSize": 16,
            },
          ]
        }
      >
        Labore sunt
      </Text>
    </View>
    <View
      animatedStyle={
        {
          "value": {
            "transform": [
              {
                "translateX": 0,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        {
          "flexDirection": "row",
          "gap": 12,
          "transform": [
            {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#1B1D4D",
              "borderRadius": 100,
              "height": 8,
              "width": 8,
            },
            {
              "opacity": 0.2,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#1B1D4D",
              "borderRadius": 100,
              "height": 8,
              "width": 8,
            },
            {
              "opacity": 0.2,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#1B1D4D",
              "borderRadius": 100,
              "height": 8,
              "width": 8,
            },
            {
              "opacity": 0.2,
            },
          ]
        }
      />
      <View
        style={
          [
            {
              "backgroundColor": "#1B1D4D",
              "borderRadius": 100,
              "height": 8,
              "width": 8,
            },
            {
              "opacity": 0.2,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "gap": 10,
          "justifyContent": "center",
          "width": 519.975,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            2
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            4
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            5
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            6
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            7
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            8
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            9
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": "auto",
              "padding": 10,
              "width": "100%",
            },
            {
              "elevation": 20,
              "shadowColor": "#16196670",
              "shadowOffset": {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            {
              "alignItems": "center",
              "height": 70,
              "justifyContent": "center",
              "width": 80,
            },
          ]
        }
        testID="universal-container"
      >
        <SvgMock />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "white",
                "borderRadius": 10,
                "height": "auto",
                "padding": 10,
                "width": "100%",
              },
              {
                "elevation": 20,
                "shadowColor": "#16196670",
                "shadowOffset": {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.08,
                "shadowRadius": 16,
              },
              {
                "alignItems": "center",
                "height": 70,
                "justifyContent": "center",
                "width": 80,
              },
            ]
          }
          testID="universal-container"
        >
          <Text
            style={
              [
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 16,
                },
                {
                  "color": "#1B1D4D",
                  "fontFamily": "MulishRegular",
                  "fontSize": 24,
                },
              ]
            }
          >
            0
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": "auto",
              "padding": 10,
              "width": "100%",
            },
            {
              "elevation": 20,
              "shadowColor": "#16196670",
              "shadowOffset": {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.08,
              "shadowRadius": 16,
            },
            {
              "alignItems": "center",
              "height": 70,
              "justifyContent": "center",
              "width": 80,
            },
          ]
        }
        testID="universal-container"
      >
        <SvgMock />
      </View>
    </View>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          [
            {
              "color": "#FF5887",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            undefined,
          ],
        ]
      }
    >
      Lost your password?
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          [
            {
              "color": "#FF5887",
              "fontFamily": "MulishRegular",
              "fontSize": 16,
            },
            undefined,
          ],
        ]
      }
    >
      Switch user
    </Text>
  </View>
</View>
`;
